feat(evm): support multiple languages for deriveKey cheatcode

0xYYY opened the pull request:
<!--
Thank you for your Pull Request. Please provide a description above and review
the requirements below.

Bug fixes and new features should include tests.
-->

## Motivation

<!--
Explain the context and why you're making that change. What is the problem
you're trying to solve? In some cases there is not a problem and this can be
thought of as being the motivation for your change.
-->

Resolves #4560.

## Solution

<!--
Summarize the solution and provide any necessary context needed to understand
the code change.
-->
Add support for other languages' wordlist in https://github.com/summa-tx/bitcoins-rs/pull/119/.

mattsse commented on evm/src/executor/inspector/cheatcodes/util.rs:
```diff
@@ -146,9 +152,41 @@ pub fn apply<DB: Database>(
         }
-        HEVMCalls::DeriveKey1(inner) => derive_key(&inner.0, &inner.1, inner.2),
+        HEVMCalls::DeriveKey1(inner) => derive_key::<English>(&inner.0, &inner.1, inner.2),
+        HEVMCalls::DeriveKey2(inner) => match inner.2.as_str() {
```
I'd like to have this as standalone function `derive_key_with_wordlist` that accepts the wordlist as str. perhaps even worth having an enum for this

mattsse reviewed (changes requested):
nice!

nit re str matching

prestwich commented:
`coins-bip39@0.8.4` contains non-english wordlists. Thanks @0xYYY!

0xYYY commented:
pending another fix https://github.com/summa-tx/coins/pull/120

DaniPopes commented on Cargo.toml:
```diff
@@ -60,15 +60,15 @@ codegen-units = 1
+ethers-signers = { git = "https://github.com/0xYYY/ethers-rs", branch = "chore/bump-bitcoin-rs" }
+ethers-middleware = { git = "https://github.com/0xYYY/ethers-rs", branch = "chore/bump-bitcoin-rs" }
+ethers-etherscan = { git = "https://github.com/0xYYY/ethers-rs", branch = "chore/bump-bitcoin-rs" }
+ethers-solc = { git = "https://github.com/0xYYY/ethers-rs", branch = "chore/bump-bitcoin-rs" }
```
We don't depend on ethers git anymore, can this be in a new release?

Evalir commented:
hey @0xYYY — do you plan on getting back to this soon?

0xYYY commented:
> hey @0xYYY — do you plan on getting back to this soon?

yeah, plan to finish this around next week

0xYYY commented:
added a standalone `derive_key_with_wordlist ` function, bumped ethers version, rebased on master, all tests have passed, ready for review

cc @Evalir @mattsse

Evalir reviewed (approved):
this lgtm—deferring to matt for final approval. we could maybe add a simple failing test to ensure it fails as expected?

0xYYY commented:
> hmm seems we have some tests failing?

yeah, but weirdly can't see any details about why the tests failed

<img width="1436" alt="Screenshot 2023-05-10 at 23 01 38" src="https://github.com/foundry-rs/foundry/assets/86655648/80b265e8-509d-42a0-b759-056b985ee007">

0xYYY commented:
> hmm seems we have some tests failing?

seems like the failing tests have ben re-ran and all tests have passed now

Evalir commented:
yup guessed they were just outage/unrelated failures, nice! will wait for matt to merge—nice work :)

Evalir commented:
hey @0xYYY — do you think we could rebase this? sorry for the extra work, but we recently refactored the abigen part of cheatcodes so will need to redo that bit

0xYYY commented:
> hey @0xYYY — do you think we could rebase this? sorry for the extra work, but we recently refactored the abigen part of cheatcodes so will need to redo that bit

no prob, on it

Evalir reviewed (approved):
aight this seems ok now—just seems we have an unrelated ci failure due to a flaky test

0xYYY commented:
>  will wait for matt to review/merge—nice work :)

hi @mattsse, pls let me know if there are any changes that i can make to push forward this pr, thx

Evalir reviewed (approved):
This is good, but will need a rebase & a redo on the abi crate prob. will do so tomorrow and get this over the line cc @mattsse

0xYYY commented:
thanks @Evalir for pushing this over the line
